xcb-types.el

```diff
@@ -456,6 +462,66 @@ Consider let-bind it rather than change its global value."))
             (throw 'break 'nil)))))
     result))
 
+
+(defsubst xcb:-f64-to-binary64 (value)
+  "Encode a 64-bit float VALUE as a binary64 (IEEE 754)."
+  (let* ((sigexp (frexp value))
+         (exp (+ (cdr sigexp) 1022))
+         (frac (abs (car sigexp)))
+         (isneg (< (copysign 1.0 (car sigexp)) 0)) ; use `copysign' to detect -0.0
+         (signmask (if isneg #x8000000000000000 0)))
+    (+ (cond ((zerop frac) 0)                                          ; 0
+             ((isnan frac) #xff0000000000001)                          ; NaN
+             ((or (>= exp 2047) (= frac 1e+INF)) #x7ff0000000000000)   ; Inf
+             ((<= exp 0) (ash (round (ldexp frac 52)) exp))            ; Subnormal
+             (t (+ (ash exp 52) (logand #xfffffffffffff
+                                        (round (ldexp frac 53))))))    ; Normal
+       signmask)))
+
+(defsubst xcb:-f32-to-binary32 (value)
```

Comment:
Just bikeshedding, but is binary32 commonly used as name for unpacked f32? We could also use unpack-f32 and pack-f32 to convert back and forth.

Reply:
1. Naming: that's what IEEE 754 calls them call them.
2. Defsubst: There isn't a single defun in the entire file. IMO, we should be using more defuns, but I'd like to stick to the convention for now.

Reply:
Thanks, all good!